### DISCUSSION TITLE
Response Parsers

### DISCUSSION THREAD

**Discussion by aacebo** [microsoft/teams-sdk#655] [Ideas / Feature requests]
## Problem
One of the biggest issues with LLM usage is that it responds with unstructured data. For example:
![image](https://github.com/microsoft/teams-ai/assets/44811138/85fd651f-70e1-4108-891a-5f3f50791666)

As you can see the it responded with some plain text, then the JSON we wanted wrapped in markdown. This poses a problem for the developer because now they have the data they requested but in a difficult to use format.

## Solution
A set of builtin response parsers that will automatically parse formatted data out of the LLM response, initial supported formats:
- JSON
- markdown
- HTML
- YAML
- URL

**Comment by singhk97** [comment]
[TypeChat](https://github.com/microsoft/TypeChat) is a library that maps LLM response to user-defined schemas. It would be nice to leverage this to solve our developer scenarios.

**Comment by Canarava** [comment]
@aacebo @singhk97
I'm sure this improvement will be heavily appreciated by a lot of developers (including me).
Any rough estimation on timeline when this improvement would be integrated?

**Comment by Stevenic** [comment]
The JS version of the AI library has a fuzzy JSON response parser which address this. OpenAI also has a new feature which lets you force JSON responses from the model.
